govcd/vapp.go

```diff
@@ -124,10 +124,8 @@ func (vapp *VApp) AddVM(orgVdcNetworks []*types.OrgVDCNetwork, vappNetworkName s
 			},
 			InstantiationParams: &types.InstantiationParams{
 				NetworkConnectionSection: &types.NetworkConnectionSection{
-					Type:                          vappTemplate.VAppTemplate.Children.VM[0].NetworkConnectionSection.Type,
-					HREF:                          vappTemplate.VAppTemplate.Children.VM[0].NetworkConnectionSection.HREF,
 					Info:                          "Network config for sourced item",
-					PrimaryNetworkConnectionIndex: vappTemplate.VAppTemplate.Children.VM[0].NetworkConnectionSection.PrimaryNetworkConnectionIndex,
+					PrimaryNetworkConnectionIndex: 0,
```

Comment:
Curious, why are we able to specify zero now, but previously could not?

Reply:
If I remember @dn1s changed this so that we do not hit empty index when a vApp template does not have network cards assigned.

Reply:
And is it safe to assume that zero is the primary network connection? I'm asking because I recall elsewhere in this PR there was a comment that the first connection is not guaranteed to be the primary.

Reply:
@lvirbalas you are right first connection may not be the primary connection. This is just a placeholder for first vm spin up. Network config will be changed later to the correct value of "PrimaryNetworkConnectionIndex". This is a fix for the problem if there is no network attached to a template.

Reply:
Yep. Actually such flow (first create with 'any' parameters and then update to correct ones) seems to be flawed and I suspect that https://github.com/terraform-providers/terraform-provider-vcd/issues/49 and https://github.com/terraform-providers/terraform-provider-vcd/issues/216 are about similar thing. I wanted to leave this PR without these fixes just to finally merge it, but  I'll try to see if we could actually create  network cards with proper configs from beginning instead of using two step approach (create anything, update to correct settings).

Reply:
Thank you and hope it won't cause too much changes. I'll keep this thread alive to share your findings in.